recipes/enchant/all/patches/0006-add-macos-support.patch

```diff
@@ -0,0 +1,51 @@
+--- lib/relocatable.c
++++ lib/relocatable.c
+@@ -37,6 +37,10 @@
+ #include <stdlib.h>
+ #include <string.h>
+ 
++#ifdef HAVE_DLADDR
++# include <dlfcn.h>
++#endif
++
+ #ifdef NO_XMALLOC
+ # define xmalloc malloc
+ #else
+@@ -389,6 +393,19 @@
+   return 1;
+ }
+ 
++#elif defined HAVE_DLADDR
++
++static Dl_info lib_info;
++
++static void
++find_shared_library_fullname (void)
++{
++  if (dladdr (&lib_info, &lib_info))
++    {
++      shared_library_fullname = strdup (lib_info.dli_fname);
```

Comment:
IIRC `strdup` allocates memory, is it freed, or leaked?

Reply:
[Chesterton's Fence](https://www.chesterton.org/taking-a-fence-down/): all the other methods set the variable to a string on the heap, so I did the same.

Reply:
But to answer your question, it's allocated only once and never free'd manually.